test/muxer/integration_test.exs

```diff
@@ -0,0 +1,66 @@
+defmodule Membrane.MPEGTS.Muxer.IntegrationTest do
+  @moduledoc false
+  use ExUnit.Case, async: true
+
+  import Membrane.Testing.Assertions
+  import Membrane.ChildrenSpec
+
+  alias Membrane.Pipeline
+  alias Membrane.MPEGTS.Muxer
+  alias Membrane.Testing.Pipeline
+
+  @input_audio "test/fixtures/bbb.aac"
+  @input_video "test/fixtures/bbb.h264"
+  @ref_audio_and_video "test/fixtures/integration/reference_with_audio_and_video.ts"
+  @ref_only_video "test/fixtures/integration/reference_only_video.ts"
+
+  @tag :tmp_dir
+  test "muxer muxes audio and video streams", ctx do
+    result_path = Path.join(ctx.tmp_dir, "result.ts")
+
+    spec = [
+      child(:audio_source, %Membrane.File.Source{location: @input_audio})
+      |> child(:audio_parser, Membrane.AAC.Parser)
+      |> via_in(:audio_input)
+      |> child(:muxer, Muxer)
+      |> child(:sink, %Membrane.File.Sink{location: result_path}),
+      child(:video_source, %Membrane.File.Source{location: @input_video})
+      |> child(:video_parser, %Membrane.H264.Parser{
+        generate_best_effort_timestamps: %{framerate: {25, 1}}
+      })
+      |> via_in(:video_input)
+      |> get_child(:muxer)
+    ]
+
+    pid = Pipeline.start_link_supervised!()
+    Pipeline.execute_actions(pid, spec: spec)
+
+    assert_end_of_stream(pid, :sink)
+    Pipeline.terminate(pid)
+
+    assert File.read!(result_path) == File.read!(@ref_audio_and_video)
```

Comment:
Are you sure this type of assertion won't produce false negatives? I can imagine, that muxer sometimes will put the audio first and sometimes it will put the video first, and both behaviours are correct

Reply:
As discussed, I haven't encountered that, let's leave it this way and we can think of a different testing approach if it ever becomes a problem